jwql/utils/utils.py

```diff
@@ -33,13 +33,108 @@
 import json
 import os
 import re
+import shutil
 
 from jwql.utils import permissions
 from jwql.utils.constants import FILE_SUFFIX_TYPES, JWST_INSTRUMENT_NAMES_SHORTHAND
 
 __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
 
 
+def copy_files(files, out_dir):
+    """Copy a given file to a given directory. Only try to copy the file
+    if it is not already present in the output directory.
+
+    Parameters
+    ----------
+    files : list
+        List of files to be copied
+
+    out_dir : str
+        Destination directory
+
+    Returns
+    -------
+    success : list
+        Files successfully copied (or that already existed in out_dir)
+
+    failed : list
+        Files that were not copied
+    """
+    # Copy files if they do not already exist
+    success = []
+    failed = []
+    for input_file in files:
+        input_new_path = os.path.join(out_dir, os.path.basename(input_file))
+        if os.path.isfile(input_new_path):
+            success.append(input_new_path)
+        else:
+            try:
+                shutil.copy2(input_file, out_dir)
+                success.append(input_new_path)
+            except:
+                failed.append(input_file)
+    return success, failed
+
+
+def download_mast_data_via_astroquery(query_results, output_dir):
+    """TEST"""
+    pass
+
+
+def download_mast_data(query_results, output_dir):
+    """Example function for downloading MAST query results. From MAST website
+    https://mast.stsci.edu/api/v0/pyex.html
+
+    Parameters
+    ----------
+    query_results : list
+        List of dictionaries returned by a MAST query.
+
+    output_dir : str
+        Directory into which the files will be downlaoded
+    """
+    # Set up the https connection
+    server = 'mast.stsci.edu'
+    conn = httplib.HTTPSConnection(server)
+
+    # Dowload the products
+    print('Number of query results: {}'.format(len(query_results)))
```

Comment:
@bhilbert4 Since this function will usually be used for instrument monitors, it would be a good idea to change the `print` statements in this module to `logging.info()` statements.

Reply:
Done